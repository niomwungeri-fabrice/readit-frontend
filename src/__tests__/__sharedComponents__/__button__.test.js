
import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import Button from "../../components/common/Buttons/BasicButton";

const setUp = props => shallow(<Button {...props} />);
const props = {
  className: "btn btn-lock",
  title: "Submit",
  onClick: jest.fn()
};
describe("Basic button", () => {
  let component;
  beforeEach(() => {
    component = setUp(props);
  });

  test("should match the snapshot", () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  test("should render button", () => {
    expect(component.hasClass(props.className)).toBe(true);
    expect(component.props().onClick).toBeDefined();
    component.simulate("click");
    expect(component.props().onClick).toHaveBeenCalledTimes(1);
  });
});
