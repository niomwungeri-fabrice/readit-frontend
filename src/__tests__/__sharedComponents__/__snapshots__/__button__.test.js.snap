// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic button should match the snapshot 1`] = `
<button
  className="btn btn btn-lock"
  onClick={[MockFunction]}
  type="button"
>
  Submit
</button>
`;
